packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -364,23 +460,39 @@ export class OrderedClientElection
     }
 
     public incrementElectedClient(sequenceNumber: number): void {
-        const nextClient = this.findFirstEligibleClient(this._electedClient?.youngerClient);
-        this.tryElectingClient(nextClient, sequenceNumber);
+        const nextClient = this.findFirstEligibleParent(this._electedParent?.youngerClient) ??
+            this.findFirstEligibleParent(this.orderedClientCollection.oldestClient);
+        if (this._electedClient === undefined || this._electedClient === this._electedParent) {
+            this.tryElectingClient(nextClient, sequenceNumber);
+        }
+        else {
+            // The _electedClient is a summarizer and should not be replaced until it leaves the quorum.
+            // Changing the _electedParent will stop the summarizer.
+            this.tryElectingParent(nextClient, sequenceNumber);
+        }
     }
 
     public resetElectedClient(sequenceNumber: number): void {
-        const firstClient = this.findFirstEligibleClient(this.orderedClientCollection.oldestClient);
-        this.tryElectingClient(firstClient, sequenceNumber);
+        const firstClient = this.findFirstEligibleParent(this.orderedClientCollection.oldestClient);
```

Comment:
This and the above function are almost similar except how they find the parent. What's the difference between the two?

Reply:
incrementElectedClient starts its search at the next-oldest client after electedParent, on the assumption that all the older ones are bad. resetElectedClient ignores electedParent and goes directly to the oldest client in the quorum.

Reply:
I see. It looks like if there is no elected client, resetElectedClient is called to choose the oldest client as summarizer. But if for some reason, the elected client is misbehaving (like it has not summarized for quite some time), incrementElectedClient is called.
nit: It would be nice to add comments for these functions to explain these :-).